packages/modules/vehicles/polestar/api.py

```diff
@@ -92,7 +89,11 @@ def check_vin(self) -> None:
 def fetch_soc(user_id: str, password: str, vin: str, vehicle: int) -> CarState:
     api = PolestarApi(user_id, password, vin)
     bat_data = api.get_battery_data()
-    soc = bat_data['batteryChargeLevelPercentage']
-    est_range = bat_data['estimatedDistanceToEmptyKm']
+    # preset values will be returned if api fails
+    soc = 0
+    est_range = 0
+    if bat_data is not None:
+        soc = bat_data['batteryChargeLevelPercentage']
+        est_range = bat_data['estimatedDistanceToEmptyKm']
```

Comment:
```suggestion
    soc = bat_data['batteryChargeLevelPercentage']
    est_range = bat_data['estimatedDistanceToEmptyKm']
```
Default-Werte werden zentral gesetzt.

Reply:
Was heisst zentral? Das geht doch über die fetch_soc Funktion und die muss doch irgendwelche Werte übergeben. Auch hier soll eine unnötige Exception (und ein überflüssiger log Eintrag) vermieden werden

Reply:
In welchen Fällen ist bat_data == None? Wenn die Abfrage fehlgeschlagen ist. Und dieser Fehlschlag soll geloggt werden und dem Benutzer im UI angezeigt werden, damit er weiß, dass die SoC-Abfrage fehlgeschlagen ist.
Das setzt voraus, dass die fetch_soc-Funktion die Exception nicht abfängt und die Werte nicht mit einem Standart-Wert füllt.
Der Zweig von `if bat_data is not None:` sollte nie erreicht werden, denn in diesem Fall muss die Funktion mit einer Exception verlassen worden sein, damit die Fehlerbehandlung das Loggen der Exception und die Meldung für das UI durchführen kann.

Reply:
> Auch hier soll eine unnötige Exception (und ein überflüssiger log Eintrag) vermieden werden

Was meinst Du mit unnötig? Wenn die Abfrage fehlschlägt, ist die Exception doch nicht unnötig, sondern nötig.

Reply:
Ich hatte das noch nicht verstanden, dass die Funktion dann verlassen wird